bridge-adapters/kinetic-bridgehub-adapter-generic-rest/src/main/java/com/kineticdata/bridgehub/adapter/generic/rest/GenericRestAdapter.java

```diff
@@ -0,0 +1,416 @@
+        new ConfigurableProperty(Properties.PROPERTY_ORIGIN).setIsRequired(true)
+            .setDescription("The scheme://hostname:port of the Ars Server")
+    );
+
+    // Local variables to store the property values in
+    private String username;
+    private String password;
+    private String origin;
+    private GenericRestQualificationParser parser;
+    private GenericRestApiHelper apiHelper;
+    
+    /*---------------------------------------------------------------------------------------------
+     * SETUP METHODS
+     *-------------------------------------------------------------------------------------------*/
+
+    @Override
+    public void initialize() throws BridgeError {
+        // Initializing the variables with the property values that were passed
+        // when creating the bridge so that they are easier to use
+        username = properties.getValue(Properties.PROPERTY_USERNAME);
+        password = properties.getValue(Properties.PROPERTY_PASSWORD);
+        origin = properties.getValue(Properties.PROPERTY_ORIGIN);
+        
+        apiHelper = new GenericRestApiHelper(origin, username, password);
+    }
+
+    @Override
+    public String getName() {
+        return NAME;
+    }
+
+    @Override
+    public String getVersion() {
+       return VERSION;
+    }
+
+    @Override
+    public void setProperties(Map<String,String> parameters) {
+        // This should always be the same unless there are special circumstances
+        // for changing it
+        properties.setValues(parameters);
+    }
+
+    @Override
+    public ConfigurablePropertyMap getProperties() {
+        // This should always be the same unless there are special circumstances
+        // for changing it
+        return properties;
+    }
+
+    /*---------------------------------------------------------------------------------------------
+     * IMPLEMENTATION METHODS
+     *-------------------------------------------------------------------------------------------*/
+
+    @Override
+    public Count count(BridgeRequest request) throws BridgeError {
+        // Log the access
+        LOGGER.trace("Counting records");
+        LOGGER.trace("  Structure: " + request.getStructure());
+        LOGGER.trace("  Query: " + request.getQuery());
+        
+        // Call common function for all BridgeAdpater methods. The method manages:
+        // parameter replacment for qualification, removing the json root, splitting
+        // the partial path from query, building partial url and calling the api
+        // helper.
+        JSONArray jsonArray = getResultsArray(request);
+
+        // Create and return a count object that contains the count
+        return new Count(jsonArray.size());
+    }
+
+    @Override
+    public Record retrieve(BridgeRequest request) throws BridgeError {
+        // Log the access
+        LOGGER.trace("Retrieving Kinetic Request CE Record");
```

Comment:
Again, should not reference a specific system. Change to something like `Retrieving Record`.